lib/topic_view.rb

```diff
@@ -735,6 +721,67 @@ def read_posts_set
 
   private
 
+  def load_mentioned_users
+    mentions = load_mentions
+    users = load_users(mentions.values.flatten.uniq)
+
+    @mentioned_users = mentions.map do |post_id, user_ids|
+      mentioned_users = user_ids.map { |id| users[id] }.compact
+      [post_id, mentioned_users]
+    end.to_h
+  end
+
+  def load_users(ids)
+    users = User.where(id: ids)
+    users = users.includes(:user_status) if SiteSetting.enable_user_status
+    users.to_h { |u| [u.id, u] }
+  end
+
+  def load_mentions
+    post_ids = @posts.map(&:id)
+    mentions_by_post = Discourse.redis.hmget("post_mentions", post_ids)
+
+    #
+    # here's what redis.hmget returns
+    #
+    # [0] = ""    – there is no mentions in this post
+    # [1] = "1,2" – users with ids 1 and 2 are mentioned in this post
+    # [2] = nil   – there is no cache for this post yet
+    #
+
+    mentions_from_cache = {}
+    parsed_mentions = {}
+    mentions_by_post.each_with_index do |value, index|
+      post_id = post_ids[index]
+      if value.nil?
+        usernames = @posts[index].parse_mentioned_usernames
+        parsed_mentions[post_id] = usernames
```

Comment:
Probably don't need that temporary variable?

```suggestion
        parsed_mentions[post_id] = @posts[index].parse_mentioned_usernames
```